warp/Network/Wai/Handler/Warp.hs

```diff
@@ -371,3 +391,10 @@ setHTTP2Disabled y = y { settingsHTTP2Enabled = False }
 -- Since 3.0.10
 pauseTimeout :: Request -> IO ()
 pauseTimeout = fromMaybe (return ()) . Vault.lookup pauseTimeoutKey . vault
+
+-- | Getting file information from the inside cache.
+--   If file information is not available, an IO error is thrown.
+--
+-- Since 3.1.10
+getFileInfo :: Request -> FilePath -> IO FileInfo
+getFileInfo = fromMaybe (\_ -> throwIO (userError "getFileInfo")) . Vault.lookup getFileInfoKey . vault
```

Comment:
What I'm saying is that there are really two different ways this can fail:

1. The `Vault.lookup getFileInfoKey` fails. In this case, it's because Warp isn't being used, and I'm OK with that throwing an exception, since it's a clear misuse of the API
2. The lookup function itself fails because the file in question isn't in the cache. That's the case that I'd prefer not throw an exception, but instead return a `Nothing` value.